### PR TITLE
Replace resque with sidekiq.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
We heart sidekiq.

part of #1888, which has instructions for deployment

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_preassembly_image_spec.rb) on stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

**Comment by jmartin-sul** [comment]
test plan (from meeting 2022-11-04):
- [x] deploy to stage (done 2022-11-10)
  - [ ] will re-base and re-deploy as needed follow testing of dependency updates tag
- [ ] Testing will proceed for 1-2 weeks.
- [ ] Test: Determine which druids have been processed and run checksum validation and replication audit.
- [ ] Test: Manually trigger catalog to moab and moab to catalog.
- [ ] Test: Re-deploy the sidekiq branch while sidekiq is running (to simulate weekly dependency updates).
- [ ] Test (Andrew): Run multiple large accessioning jobs.

**Review by jmartin-sul — COMMENTED** [review]
A question for Tony and a thought (maybe for a follow-on PR) about the okcomputer checks, but looks good to me pending successful testing!

File: config/routes.rb [hunk 1]
```diff
@@ -1,11 +1,8 @@
 # frozen_string_literal: true
 
-require 'resque/server'
-
 Rails.application.routes.draw do
-  mount Resque::Server.new,
-        at: '/resque',
-        constraints: ->(req) { Settings.resque_dashboard_hostnames.include?(req.host) }
```
[the remark this answers is not in the document]
Comment (tallenaz) [reply config/routes.rb:8]: I'm hazy on why we decided to limit it to fewer hosts, but I don't have any objection to it at more hosts.

[the remark this answers is not in the document]
Comment (jmartin-sul) [reply config/routes.rb:8]: cool, thanks for weighing in!

**Comment by jmartin-sul** [comment]
ref #1888

**Comment by jmartin-sul** [comment]
for possible investigation later: these stage druids, first preserved on the date listed, were mentioned in honeybadger alerts for the replication audits i triggered last night, which i ran on all druids accessioned on stage in the last 30 days, according to an argo CSV report (generated from roughly [this search](https://argo-stage.stanford.edu/catalog?all=true&f[accessioned_latest_dttsi][]=[2022-11-10T00%3A00%3A00.000Z+TO+2022-12-08T12%3A00%3A00.000Z]&commit=submit)):

```ruby
[["zp159nr3578", Wed, 16 Nov 2022 13:30:35.417790000 UTC +00:00],
 ["dz927kn8537", Thu, 17 Nov 2022 20:40:27.857617000 UTC +00:00],
 ["kr909fz1118", Sun, 20 Nov 2022 20:45:22.200051000 UTC +00:00],
 ["dp541ph9411", Mon, 21 Nov 2022 21:45:36.050225000 UTC +00:00],
 ["nx435db4242", Sun, 27 Nov 2022 15:15:23.356342000 UTC +00:00],
 ["jy953bv8624", Tue, 29 Nov 2022 22:30:30.409274000 UTC +00:00],
 ["tt509dc8788", Wed, 07 Dec 2022 21:46:00.888965000 UTC +00:00],
 ["vm797nd4632", Thu, 08 Dec 2022 11:00:34.988473000 UTC +00:00]]
````

full list of stage druids i audited:

[prescat-sidekiq-testing_stage-druids-accessioned_2022-11-10_to_12-08_argo-report.csv](https://github.com/sul-dlss/preservation_catalog/files/10190879/prescat-sidekiq-testing_stage-druids-accessioned_2022-11-10_to_12-08_argo-report.csv)

andrew's large test objects:
https://argo-stage.stanford.edu/view/druid:kk167wy1190 (179 GB)
https://argo-stage.stanford.edu/view/druid:gt555nw4813 (179 GB)

when i run the manual/synchronous replication debug code from this wiki entry on those druids, all's well:
https://github.com/sul-dlss/preservation_catalog/wiki/Investigating-a-druid-with-replication-errors/33802d5dd3fb7077c5de416dbba917ec23e88fc8

** bucket info:  #<Aws::S3::Bucket:0x0000564fa08d3740 @name="sul-sdr-aws-us-east-1-test", @data=nil, @client=#<Aws::S3::Client>, @waiter_block_warned=false>
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z01 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z02 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z03 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z04 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z05 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z06 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z07 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z08 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z09 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z10 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z11 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z12 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z13 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z14 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z15 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.zip Found = true
** bucket info:  #<Aws::S3::Bucket:0x0000564f9fa09a48 @name="sul-sdr-aws-us-west-2-test", @data=nil, @client=#<Aws::S3::Client>, @waiter_block_warned=false>
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z01 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z02 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z03 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z04 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z05 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z06 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z07 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z08 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z09 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z10 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z11 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z12 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z13 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z14 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z15 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.zip Found = true
** bucket info:  #<Aws::S3::Bucket:0x0000564f9f409350 @name="sul-sdr-ibm-us-south-1-test", @data=nil, @client=#<Aws::S3::Client>, @waiter_block_warned=false>
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z01 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z02 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z03 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z04 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z05 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z06 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z07 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z08 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z09 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z10 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z11 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z12 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z13 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z14 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.z15 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0001.zip Found = true
** bucket info:  #<Aws::S3::Bucket:0x0000564fa05fafc8 @name="sul-sdr-aws-us-east-1-test", @data=nil, @client=#<Aws::S3::Client>, @waiter_block_warned=false>
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.z01 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.z02 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.zip Found = true
** bucket info:  #<Aws::S3::Bucket:0x0000564fa0b92c60 @name="sul-sdr-aws-us-west-2-test", @data=nil, @client=#<Aws::S3::Client>, @waiter_block_warned=false>
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.z01 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.z02 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.zip Found = true
** bucket info:  #<Aws::S3::Bucket:0x0000564f9c8972a8 @name="sul-sdr-ibm-us-south-1-test", @data=nil, @client=#<Aws::S3::Client>, @waiter_block_warned=false>
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.z01 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.z02 Found = true
   ** part info:  Key = kk/167/wy/1190/kk167wy1190.v0002.zip Found = true

**Comment by jmartin-sul** [comment]
did some auditing on the last few hours of prod traffic, things seem good there.  i ran [this argo search](https://argo.stanford.edu/report?all=true&f%5Baccessioned_latest_dttsi%5D%5B%5D=%5B2022-12-08T20%3A00%3A00.000Z+TO+2022-12-09T00%3A45%3A00.000Z%5D) and downloaded the CSV report (attached) to get everything from noon pacific/8 pm UTC today, till about 4:45 pm pacific.  deployment was at 1:07 pm pacific, so about an hour before, and just under 4 hours after.

i confirmed that all of the druids existed in preservation at the same version as argo reported, and then ran checksum validation and replication audit on them.  no errors detected among any of those checks.  i watched sidekiq while those audit jobs were queued and after they started to get worked, and they finished in the couple minutes during which i fumbled around to get the search link i used for my report (i hadn't put an end time on it when i ran it, so i had to futz around a bit to get a repeatable link with the same results).

```ruby
3.0.0 :004 > accessioned_csv = CSV.read('/opt/app/presprescat-sidekiq_druids-accessioned-since-1hr-before-deployment_2022-12-08T20-00-00Z_to_2022-12-00T00-30-00Z_argo-report.csv', headers: true)
 => #<CSV::Table mode:col_or_row row_count:787> 
3.0.0 :005 > 
3.0.0 :006 > accessioned_csv.by_col['Druid'].first
 => "bb677kq1601" 
3.0.0 :007 > 
3.0.0 :008 > accessioned_csv.by_col['Druid'].reject { |druid| PreservedObject.exists?(druid: druid) }
 => [] 
3.0.0 :009 > 
3.0.0 :010 > accessioned_csv.reject { |row| row['Status'].start_with?("v#{PreservedObject.find_by!(druid: row['Druid']).current_version} ") }
 => [] 
3.0.0 :011 > accessioned_csv.by_col['Druid'].each { |druid| CompleteMoab.by_druid(druid).first.validate_checksums!; PreservedObject.find_by!(druid: druid).audit_moab_version_replication! }
3.0.0 :012 > 
3.0.0 :013 > accessioned_csv.size
 => 786 
```

No audit errors reported to Honeybadger, no job failures 🎉 

[prescat-sidekiq_druids-accessioned-since-1hr-before-deployment_2022-12-08T20-00-00Z_to_2022-12-00T00-30-00Z_argo-report.csv](https://github.com/sul-dlss/preservation_catalog/files/10190988/prescat-sidekiq_druids-accessioned-since-1hr-before-deployment_2022-12-08T20-00-00Z_to_2022-12-00T00-30-00Z_argo-report.csv)

**Review by jmartin-sul — APPROVED** [review]
seems this has been doing fine in prod overnight, code was reviewed a while back and seemed fine.  so i think we're moving forward, and will troubleshoot/tune performance issues as needed.  seems unlikely we'll roll back to resque.

✅ :shipit:

**Pull request merged**